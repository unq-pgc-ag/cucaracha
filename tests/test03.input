
fun producto(v : Vec) : Int {
  i := 0
  p := 1
  while i < #v {
    p := p * v[i]
    i := i + 1
  }
  return p
}

fun main() {
  x := [5,4,3,2,1]
  putNum(producto(x))
}

