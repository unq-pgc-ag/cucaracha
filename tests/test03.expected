(Program
  (Function
    producto
    Int
    (Parameter
      v
      Vec
    )
    (Block
      (StmtAssign
        i
        (ExprConstNum
          0
        )
      )
      (StmtAssign
        p
        (ExprConstNum
          1
        )
      )
      (StmtWhile
        (ExprLt
          (ExprVar
            i
          )
          (ExprVecLength
            v
          )
        )
        (Block
          (StmtAssign
            p
            (ExprMul
              (ExprVar
                p
              )
              (ExprVecDeref
                v
                (ExprVar
                  i
                )
              )
            )
          )
          (StmtAssign
            i
            (ExprAdd
              (ExprVar
                i
              )
              (ExprConstNum
                1
              )
            )
          )
        )
      )
      (StmtReturn
        (ExprVar
          p
        )
      )
    )
  )
  (Function
    main
    Unit
    (Block
      (StmtAssign
        x
        (ExprVecMake
          (ExprConstNum
            5
          )
          (ExprConstNum
            4
          )
          (ExprConstNum
            3
          )
          (ExprConstNum
            2
          )
          (ExprConstNum
            1
          )
        )
      )
      (StmtCall
        putNum
        (ExprCall
          producto
          (ExprVar
            x
          )
        )
      )
    )
  )
)
